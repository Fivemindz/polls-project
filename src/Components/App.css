.dash-body {
  height: 1000px;
  display: grid;
  grid-template-rows: 200px auto;
  grid-template-columns: repeat(7, 1fr);
  grid-template-areas:
    " ban ban ban ban ban ban ban"
    "nv mn mn mn mn mn mn"
    ;
}
.container {
  margin: 10px 200px 5px 200px;
  padding: 10px;
}
.dash-loading {
  grid-area: mn;
  border: 1px solid lightgray;
  display: flex;
  align-items: center;
  justify-content: center;
}