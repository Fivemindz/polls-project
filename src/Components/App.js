import { Fragment, useEffect, useState } from "react";
import { connect } from "react-redux";
import LoadingBar from "react-redux-loading-bar";
import Login from "./Login";
import { Routes, Route } from "react-router-dom";
import Banner from "./Banner";
import Nav from "./Nav";
import Dash from "./Dash";
import Stats from "./Stats";
import NewQuestion from "./NewQuestion";
import "./App.css";
import { getUsers } from "../actions/shared";
import QuestionPage from "./QuestionPage";
import Loading from "./Loading";
import PageNotFound from "./PageNotFound";

const App = (props) => {
  const { loading, dispatch, loadingBar } = props;
  const [usersLoaded, setUsersLoaded] = useState(false);

  useEffect(() => {
    const handleLoadUsers = async () => {
      await dispatch(getUsers());
      setUsersLoaded(true);
    };

    if (usersLoaded === false) {
      handleLoadUsers();
    }

    return () => {
      setUsersLoaded(true);
    };
  });

  return (
    <Fragment>
      <LoadingBar />
      <div className="container" data-testid="home-page">
        {loading === true ? (
          <Login />
        ) : (
          <div className="dash-body">
            <Banner />
            <Nav />
            {loadingBar ? (
              <div className="dash-loading">
                <Loading />
              </div>
            ) : (
              <Routes>
                <Route path="/" exact element={<Dash />} />
                <Route path="/stats" element={<Stats />} />
                <Route path="/newquestion" element={<NewQuestion />} />
                <Route path="/question/:id" element={<QuestionPage />} />
                <Route path="/notfound" element={<PageNotFound />} />
              </Routes>
            )}
          </div>
        )}
      </div>
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  loading: authedUser === null,
});

export default connect(mapStateToProps)(App);
