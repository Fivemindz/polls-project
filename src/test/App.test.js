import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers";
import middleware from "../middleware";
import App from "../Components/App";

const store = createStore(reducer, middleware);

describe("Test App Component", () => {
  it("Snapshot:Renders the login page if user is not logged in", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  // it("Snapshot: Renders HomePage once user is logged in", async () => {
  //   const component = render(
  //     <Provider store={store}>
  //       <Router>
  //         <App />
  //       </Router>
  //     </Provider>
  //   );
  //   setTimeout(() => {
  //     const select = component.getAllByTestId("user-select");
  //     fireEvent.change(select, { target: { value: "tylermcginnis" } });
  //     var submitButton = component.getByTestId("user-login-button");
  //     fireEvent.click(submitButton);
  //     expect(component).toMatchSnapshot();
  //   }, "3000");
  // });
});
