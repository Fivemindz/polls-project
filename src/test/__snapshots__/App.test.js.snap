// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component Snapshot:Renders the login page if user is not logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <div
        class="container"
        data-testid="home-page"
      >
        <div
          class="login-page"
        >
          <div
            style="fill: blue; height: 200px; width: 75px;"
          >
            <svg
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              
  
              <path
                d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
                opacity=".25"
              />
              
  
              <path
                d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
              >
                
    
                <animatetransform
                  attributeName="transform"
                  dur="0.8s"
                  from="0 16 16"
                  repeatCount="indefinite"
                  to="360 16 16"
                  type="rotate"
                />
                
  
              </path>
              

            </svg>
            

          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="direction: ltr;"
    >
      <div
        class=""
        style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
      />
      <div
        style="display: table; clear: both;"
      />
    </div>
    <div
      class="container"
      data-testid="home-page"
    >
      <div
        class="login-page"
      >
        <div
          style="fill: blue; height: 200px; width: 75px;"
        >
          <svg
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            
  
            <path
              d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
              opacity=".25"
            />
            
  
            <path
              d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
            >
              
    
              <animatetransform
                attributeName="transform"
                dur="0.8s"
                from="0 16 16"
                repeatCount="indefinite"
                to="360 16 16"
                type="rotate"
              />
              
  
            </path>
            

          </svg>
          

        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
